packages/plugin-ext/src/hosted/browser/hosted-plugin-watcher.ts

```diff
@@ -21,7 +21,7 @@ import { LogPart } from '../../common/types';
 
 @injectable()
 export class HostedPluginWatcher {
```

Comment:
I am just slightly confused by this class `getHostedPluginClient()` method... Essentially it makes the `HostedPluginWatcher` component a factory since each call to `getHostedPluginClient()` creates a new object (which does the same thing than the previous one). Should all these events be part of the hosted client interface directly rather? And unless someone needs a custom client, it looks like it could be some kind of singleton binding?

Reply:
Or maybe just a matter of storing the object and returning the same reference everytime? Still makes me wonder if the events should be part of the client, or if it is useful to have those in the watcher. WDYT?

Reply:
You totally agree. I refactored this code in https://github.com/eclipse-theia/theia/pull/8437/files, but since that also included some major refactoring of the plugin host startup sequence, I had trouble getting it reviewed or merged.
My approach now is to get the uncontroversial parts of PR 8437 merged one by one.